docs/reference/search/profile.asciidoc

```diff
@@ -1047,6 +1046,218 @@ per hit. In that case the true cost of `_source` phase is hidden in the
+===== Profiling DFS
+
+The dfs phase collects term statistics to achieve more accurate search results
+prior to the query phase. To profile the dfs phase use a `_search` request with
+the url parameter `search_type` set as `dfs_and_then_query` and add a top-level
+`profile` parameter set to `true`. Note that for the dfs phase to run an index
+must have multiple shards.
+
+As an example, let's first setup an index with multiple shards and index
+a pair of documents with different values on a keyword field.
+
+[source,console]
+--------------------------------------------------
+PUT my-dfs-index
+{
+  "settings": {
+    "number_of_shards": 2, <1>
+    "number_of_replicas": 1
+  },
+  "mappings": {
+      "properties": {
+        "my-keyword": { "type": "keyword" }
+      }
+    }
+}
+
+POST my-dfs-index/_bulk?refresh=true
+{ "index" : { "_id" : "1" } }
+{ "my-keyword" : "a" }
+{ "index" : { "_id" : "2" } }
+{ "my-keyword" : "b" }
+--------------------------------------------------
+<1> The `my-dfs-index` is created with multiple shards.
+
+With an index setup, we can now profile the dfs phase of a
+search query. As an example we use a term query.
+
+[source,console]
+--------------------------------------------------
+GET /my-dfs-index/_search?search_type=dfs_query_then_fetch&pretty <1>
+{
+  "profile": true, <2>
+  "query": {
+    "term": {
+      "my-keyword": {
+        "value": "a"
+      }
+    }
+  }
+}
+--------------------------------------------------
+// TEST[continued]
+<1> The `search_type` url parameter is set to `dfs_query_then_fetch` to
+ensure the dfs phase is run.
+<2> The `profile` parameter is set to `true`.
+
+In the response, we see a very verbose profile which includes a `dfs` section
+for each shard along with profile output for the rest of the search phases.
+One of the `dfs` sections for a shard looks like the following:
+
+[source,js]
+--------------------------------------------------
+{
+    ...
+    "dfs" : {
+        "statistics" : {
+            "type" : "statistics",
+            "description" : "collect term statistics",
+            "time_in_nanos" : 236955,
+            "breakdown" : {
+                "term_statistics" : 4815,
+                "collection_statistics" : 27081,
+                "collection_statistics_count" : 1,
+                "create_weight" : 153278,
+                "term_statistics_count" : 1,
+                "rewrite_count" : 0,
+                "create_weight_count" : 1,
+                "rewrite" : 0
+            }
+        }
+    }
+    ...
+}
+--------------------------------------------------
+// NOTCONSOLE
+
+In the `dfs.statistics` portion of this response we see a `time_in_nanos`
+which is the total time it took to collect term statistics for this
+shard. We also see a further breakdown of times where `create_weight`
+is called to generate the collection of term statistics, and the specific
+call order is `create_weight -> collection_statistics -> term_statistics`.
```

Comment:
Maybe we can leave out this part about the call order, because Lucene doesn't guarantee it at all (for example we could easily collect term stats before collection stats). I'm also not sure it gives more insight to users.

Reply:
I removed this section completely.